code/modules/projectiles/updated_projectiles/guns/specialist.dm

```diff
@@ -44,30 +45,55 @@
 	Q.Attach(src)
 	update_attachables()
 	S.icon_state = initial(S.icon_state)
+	LT = image("icon" = 'icons/obj/items/projectiles.dmi',"icon_state" = "sniper_laser", "layer" =-LASER_LAYER)
 
 /obj/item/weapon/gun/rifle/sniper/M42A/Fire(atom/target, mob/living/user, params, reflex = 0, dual_wield)
 	if(!able_to_fire(user))
 		return
 	if(targetlaser_on)
-		if(!ismob(target))
-			return
-		if(target == laser_target)
-			to_chat(user, "<span class='warning'>You're already focused on that target!</span>")
+		if(!iscarbon(target))
 			return
 		if(laser_target)
-			laser_target.vis_contents -= laser
-		cdel(laser)
+			remove_laser()
 		to_chat(user, "<span class='danger'>You focus your targeting laser on [target]!</span>")
 		laser_target = target
-		var/obj/effect/overlay/temp/sniper_laser/LT = new (laser_target, "targeting laser")
-		laser = LT
 		targetlaser_on = FALSE
-		laser_target.vis_contents += laser
+		apply_laser()
 		processing_objects.Remove(src) //So we don't accumulate additional processing.
 		processing_objects.Add(src)
 		return
 	return ..()
 
+/obj/item/weapon/gun/rifle/sniper/M42A/proc/apply_laser()
```

Comment:
Instead of a chain of `istype` checks, do a proc for them instead.
Something like
```
/mob/living/carbon/proc/apply_laser()
	return FALSE
/mob/living/carbon/human/apply_laser()
	overlays_standing[LASER_LAYER] = image("icon" = 'icons/obj/items/projectiles.dmi',"icon_state" = "sniper_laser", "layer" =-LASER_LAYER)
	apply_overlay(LASER_LAYER)
```
And so on.